docs/HowTo/develop-subtopologies.md

```diff
@@ -0,0 +1,345 @@
+
+> Note that with the latest release of `fprime-tools`, you can run `fprime-util new --subtopology` to generate the subtopology structure.
+
+## Individual File Contents
+
+This section will provide an overview to the contents of the individual files that make up a subtopology. We will use the names of the required files structure as shown in the [above](#subtopology-structure) section. However, note that the file names shown are not required to be duplicated within your own custom subtopologies.
+
+### Example Scenario
+
+For this section, it would be helpful to come up with an example scenario where the subtopology is used, such that it can be better understood. Let's imagine that we would like to create a component called `RNG` that will write a random integer to some telemetry channel when it is hooked up to a 1Hz clock (rate group). The `RNG` component will have an input port called `run`, that will be the entry point from the clock (rate group). The `RNG` component will be an active component, and is under the `MyLibrary` namespace.
+
+### MySubtopology.fpp
+
+As mentioned, this is a unified `.fpp` file. It not only includes the instances instance declarations for components that you'd like your subtopology to use, but also the definitions for the instances.
+
+Based on our example, we may have the following instance declarations:
+
+```cpp
+instance rng // RNG component instance
+instance rateGroup // rate group instance
+``` 
+
+and thus the following instance definitions:
+
+```cpp
+instance rng: MyLibrary.RNG base id 0xFF2FF \
+    queue size Defaults.QUEUE_SIZE \
+    stack size Defaults.STACK_SIZE \
+    priority 100
+
+instance rateGroup: Svc.ActiveRateGroup base id 0xFF4FF \
+    queue size Defaults.QUEUE_SIZE \
+    stack size DEFAULTS.STACK_SIZE \
+    priority 150
+
+instance rateGroupDriver: Svc.RateGroupDriver base id 0xFF8FF // to drive the 1Hz rate group
+```
+
+and also the following wiring:
+
+```cpp
+connections MyWiring {
+    // we will hook up the cycle for our rate group later on
+    rateGroup.RateGroupMemberOut[0] -> rng.run
+}
+```
+
+Thus, your unified fpp file, and so `MySubtopology.fpp` could look like:
+
+```cpp
+module MySubtopology {
+    instance rng: MyLibrary.RNG base id 0xFF2FF \
+        queue size Defaults.QUEUE_SIZE \
+        stack size Defaults.STACK_SIZE \
+        priority 100
+
+    instance rateGroup: Svc.ActiveRateGroup base id 0xFF4FF \
+        queue size Defaults.QUEUE_SIZE \
+        stack size DEFAULTS.STACK_SIZE \
+        priority 150
+
+    topology MySubtopology {
+        instance rng // RNG component instance
+        instance rateGroup // rate group instance
+
+        connections MyWiring {
+            rateGroup.RateGroupMemberOut[0] -> rng.run
+        }
+    } // end topology
+} // end MySubtopology
+```
+
+## MySubtopologyTopology.cpp and MySubtopologyTopology.hpp (pt1)
```

Comment:
Would it read better to name the section
```
## Subtopology CPP and HPP
```
or something similar? Might be just a personal preference, let me know